### ISSUE TITLE
Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by berickson1** [WarnerHooh/babel-plugin-parameter-decorator#5]
When using this with ReSub using both parameter decorations and function decorations, we're seeing the following exception on build

```typescript
import { autoSubscribe, key, StoreBase } from 'resub';

@AutoSubscribeStore
class MyStore extends StoreBase {

    @autoSubscribe
    aFunction(@key address: string): undefined {
        // This would normally return something but not needed for an exmaple
        return undefined;
    }
}
```

error: bundling failed: TypeError: Cannot read property 'name' of undefined
    at /src/project/node_modules/babel-plugin-parameter-decorator/lib/index.js:39:60
    at Array.forEach (<anonymous>)
    at /src/project/node_modules/babel-plugin-parameter-decorator/lib/index.js:38:44
    at Array.forEach (<anonymous>)
    at /src/project/node_modules/babel-plugin-parameter-decorator/lib/index.js:37:37
    at Array.forEach (<anonymous>)
    at /src/project/node_modules/babel-plugin-parameter-decorator/lib/index.js:36:31
    at Array.forEach (<anonymous>)
    at /src/project/node_modules/babel-plugin-parameter-decorator/lib/index.js:35:16

I logged the `decorator` at line 39 and this is the output
```json
{"type":"Decorator","start":386,"end":390,"loc":{"start":{"line":13,"column":13},"end":{"line":13,"column":17}},"expression":{"type":"Identifier","start":387,"end":390,"loc":{"start":{"line":13,"column":14},"end":{"line":13,"column":17},"identifierName":"key"},"name":"key"}}
```

I think lib/index.js should probably read as
```javascript
if (decorator.expression.callee) {
    decorators[decorator.expression.callee.name] = decorator;

} else {
    decorators[deorator.expression.name] = decorator;
}
```

**Comment by WarnerHooh** [comment]
Thanks for your feedback, @berickson1 
Will have a quick look today.

**Comment by WarnerHooh** [comment]
Hello @berickson1 ,
I have fixed this and patch a new version. So this should be ok with the new version. Please let me know if you have any other problem.

**Comment by berickson1** [comment]
Thanks for the quick turnaround! This fixed the issue we were seeing

**Comment by WarnerHooh** [comment]
This issue has been resolved in `1.0.9`
